controllers/argocd/util.go

```diff
@@ -1362,6 +1364,25 @@ func (r *ReconcileArgoCD) setManagedNamespaces(cr *argoproj.ArgoCD) error {
 	return nil
 }
 
+// getSourceNamespaces retrieves a list of namespaces that match the sourceNamespaces
+// pattern specified in the given ArgoCD
+func (r *ReconcileArgoCD) getSourceNamespaces(cr *argoproj.ArgoCD) ([]string, error) {
+	sourceNamespaces := []string{}
+	namespaces := &corev1.NamespaceList{}
+
+	if err := r.Client.List(context.TODO(), namespaces, &client.ListOptions{}); err != nil {
+		return nil, err
+	}
+
+	for _, namespace := range namespaces.Items {
+		if glob.MatchStringInList(cr.Spec.SourceNamespaces, namespace.Name, false) {
+			sourceNamespaces = append(sourceNamespaces, namespace.Name)
+		}
+	}
+
+	return sourceNamespaces, nil
+}
+
 func (r *ReconcileArgoCD) setManagedSourceNamespaces(cr *argoproj.ArgoCD) error {
```

Comment:
Should this logic be updated  to call getSourceNamespaces()

Reply:
@anandf , setManagedSourceNamespaces() function currently stores the list of SourceNamespaces last updated in the ReconcileArgoCD.ManagedSourceNamespaces variable and getSourceNamespaces() return the latest ns mentioned in sourceNamespaces.

Reply:
@Mangaal , `setManagedSourceNamespaces` finds the list of namespaces that match the label `ArgoCDManagedByClusterArgoCDLabel` and set those in `ReconcileArgoCD.ManagedSourceNamespaces`.

So we need to call `getSourceNamespaces` instead of https://github.com/argoproj-labs/argocd-operator/pull/1218/files#diff-30c42e628eebe205919571b78686864b3d5aaab1d82bb09bdb56a1c53bc3d81fR1389-R1396